### PR TITLE
Status Bar

### PR DESCRIPTION
**Pull request by perunt**
Fixes RNBW-2786, RNBW-2666

## What changed (plus any additional context for devs)
Created one place where Status Bar is processed. `react-native-bootsplash` was breaking the Status Bar behavior for Android. I used `react-native-bars` from the same author. Now the status bar's behavior on Android is similar to that on iOS. I also fixed the status bar flickering when changing the theme on the settings screen and in some other places. Also removed `fromDiscover` as it is no longer needed (please correct me if this is wrong)

## PoW (screenshots / screen recordings)
Android
 - before: https://user-images.githubusercontent.com/48593211/178678218-b8701205-ccde-4c54-9135-d9ecb1f9f50e.mp4 
 - after: https://user-images.githubusercontent.com/48593211/178685248-18da37d9-d31f-4218-876e-cebec461a883.mp4

iOS
- before: https://user-images.githubusercontent.com/48593211/178679214-9b2198cf-d1e6-4162-baf0-e32a14313341.MP4
- after: https://user-images.githubusercontent.com/48593211/178679814-1e03cd97-b2ba-4613-a7f9-4bca6cd15356.mov

## Dev checklist for QA: what to test
Changing color of the Status Bar depending on screen and theme

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [ ] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -71,7 +74,7 @@ package.json @brunobar79 @estebanmino @nickbytes
 react-native.config.js @brunobar79 @estebanmino
 README.md @brunobar79 @estebanmino @nickbytes
 shim.js @brunobar79 @osdnk @estebanmino @nickbytes
-tsconfig.json @brunobar79 @osdnk @estebanmino @nickbytes
+tsconfig.json @brunobar79 @osdnk @estebanmino @nickbytes @perunt
```
Comment (nickbytes) [inline CODEOWNERS:77]: I think we should remove the addition of another codeowner to `tsconfig.json`. While I trust @perunt to make changes, it doesn't seem like `tsconfig.json` is something that will need to be updated frequently and changes should be carefully considered for their impact to the team. Having fewer codeowners for a file like this is a "feature" IMO.

**Comment by nickbytes** [comment]
Pumped to see this change! I have a few comments and questions.

This PR adds a new "pattern" to the codebase which includes a `/services` folder which contains a class to handle the Status Bar. Is there a way we can do this that is more in line with patterns already present in the codebase?

**Review by tchayen — COMMENTED** [review]
Great stuff 🤩.

Regarding what @nickbytes said, instead of that class we can have a file in utils/helpers that exports those functions (as they are not strictly tied into a class anyway).

Then call signature remains the same plus when we import it like `import * as StatusBarService from '...'`.

**Comment by osdnk** [comment]
I don't really get the idea of "Services"? That is basically a class that is a singleton, so can it be an object equally fine or even a set of functions? And then this is just a set of helpers or utils we can locate in one existing folder?

**Comment by perunt** [comment]
I dug a bit more and found that from 0.69 RN and above uses recent StatusBar APIs (https://github.com/facebook/react-native/commit/50c8e973f067d4ef1fc3c2eddd360a0709828968)
as `react-native-bootsplash` use updated StatusBar APIs, it broke it entirely for React Native < 0.69
as `react-native-bootsplash` released a new version with a fix, we can remove  `react-native-bars`. But at first, we should bump RN to 69.2
   In this case, the need to use the service will disappear, but it is still a promising approach for storing adjacent logic.
   Yes, it is unnecessary to use a class as it would be transpiled into an object anyway.
   Services will allow you to structure the code and keep similar logic in one file. For example, connecting and processing sockets, or analytics. We create a separate file that will do the dirty work under the hood and give us a friendly interface. 
   In addition, if you want to change something in these services, you can do it in one place and not look for them scattered throughout the code.
   It could make code way cleaner and understandable

File: android/app/src/main/res/values-v27/styles.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    <!-- BootTheme should inherit from Theme.SplashScreen -->
+    <style name="BootTheme" parent="Theme.SplashScreen">
+        <item name="windowSplashScreenBackground">@color/primary_dark</item>
+        <item adjustViewBounds="true" name="windowSplashScreenAnimatedIcon">@mipmap/bootsplash_logo</item>
+        <item name="postSplashScreenTheme">@style/AppTheme</item>
+        <!-- Bars initial styles: true = dark-content, false = light-content -->
+        <item name="android:windowLightStatusBar">true</item>
+    </style>
+</resources>
```
Comment (brunobar79) [inline android/app/src/main/res/values-v27/styles.xml:31]: new line

**Review by brunobar79 — COMMENTED** [review]
Not a fan of introducing new patterns. The rest looks ok to me. Flagging the minSDK version for review from others

**Comment by perunt** [comment]
with the update of React Native to version 69+, there will be no need to create this service|handler and I will remove it completely. That's why I'm sending it to draft until we have a newer version of the RN

File: android/app/src/main/res/values-v27/styles.xml (hunk 2 above)
Comment (perunt) [reply android/app/src/main/res/values-v27/styles.xml:31]: ok

File: android/build.gradle [hunk 3]
```diff
@@ -6,7 +6,7 @@ buildscript {
 
     ext {
         buildToolsVersion = "31.0.0"
-        minSdkVersion = 21
+        minSdkVersion = 23
```
[the remark this answers is not in the document]
Comment (perunt) [reply android/build.gradle]: downgraded

**Review by terrysahaidak — APPROVED** [review]
LGTM

**Review by tchayen — APPROVED** [review]
Seems to work fine in testing.

I am a bit worried though about the new `isFromWalletScreen` (and the other one) navigation params. I see why they are needed, but I wonder if we can somehow make it more discoverable to other devs in situations in which they should use them.

**Comment by tchayen** [comment]
Now it was def solved. Rebase/merge dev @perunt

**Comment by tchayen** [comment]
@perunt conflict appeared

**Review by estebanmino — APPROVED** [review]
LGTM

**Comment by terrysahaidak** [comment]
Closing in favor #4020

**Pull request closed without merging**